src/parser/Parser.ts

```diff
@@ -706,9 +706,20 @@ export class Parser {
 
             const statements: Statement[] = [];
             while (!check(...terminators) && !isAtEnd()) {
-                const dec = declaration();
+                //grab the location of the current token
+                let loopCurrent = current;
+                let dec = declaration();
+
                 if (dec) {
                     statements.push(dec);
+                } else {
+                    //something went wrong. reset to the top of the loop
+                    current = loopCurrent;
```

Comment:
Looks like this is part of the issue - when we encounter an error, it appears we rewind to the start of the line and only consume until the first `:`, causing a partial associative array to remain.